modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -323,16 +326,38 @@ class QueueReader
 void rewindToStop(std::vector<Q*> &in_queues,
                   const std::size_t  this_id)
 {
-    for (auto &&qit : ade::util::indexed(in_queues))
-    {
-        auto id2 = ade::util::index(qit);
-        auto &q2 = ade::util::value(qit);
-        if (this_id == id2) continue;
+    // Will draining all queue & waiting for stop cmd for all queues except own queue
+    size_t expected_stop_count = std::count_if(in_queues.begin(), in_queues.end(), [] (const Q* ptr) {
+        return ptr != nullptr;
+    });
 
-        Cmd cmd;
-        while (q2 && !cv::util::holds_alternative<Stop>(cmd))
-            q2->pop(cmd);
+    // NB: extract 1 to skip own queue because stop cmd from own queue truggered rewind
+    expected_stop_count = expected_stop_count > 0 ? expected_stop_count - 1: expected_stop_count;
```

Comment:
yes, it is easier

> Also, if it is 0, is it still a valid case?

I'm not sure - anyway I don't want to touch legacy behavior, because before this fix the value of 0 - was  correct value for this method and doesn't break any assert here